src/Server/HTTPHandler.cpp

```diff
@@ -108,6 +110,37 @@ namespace ErrorCodes
     extern const int HTTP_LENGTH_REQUIRED;
 }
 
+namespace
+{
+    /// Process options request. Useful for CORS.
+    void processOptionsRequest(HTTPServerResponse & response, const Poco::Util::LayeredConfiguration & config)
+    {
+        /// If there is information for options request in cofing, fill response.
+        /// For this purpose find all headers related to http_options_response and add them with their values to response
+        if (config.has("http_options_response"))
+        {
+            Strings config_keys;
+            config.keys("http_options_response", config_keys);
+            for (const std::string & config_key : config_keys)
+            {
+                if (config_key == "header" || config_key.starts_with("header["))
+                {
+                    /// If there is empty header name, it will not be processed and message about it will be in logs
+                    if (config.getString("http_options_response." + config_key + ".name", "").empty())
+                        LOG_WARNING(&Poco::Logger::get("processOptionsRequest"), "Empty header was found in config. It will not be processed.");
+                    else 
+                        response.add(config.getString("http_options_response." + config_key + ".name", ""),
+                                     config.getString("http_options_response." + config_key + ".value", ""));
+
+                }
+            }
```

Comment:
There is no exception throw, because it's up to user whether there should be headers to support CORS and which headers should be added.